source/frontend/src/features/admin/financial-codes/list/FinancialCodeListView.tsx

```diff
@@ -0,0 +1,131 @@
+import { Button } from 'components/common/buttons/Button';
+import { TableSort } from 'components/Table/TableSort';
+import { Roles } from 'constants/roles';
+import useKeycloakWrapper from 'hooks/useKeycloakWrapper';
+import orderBy from 'lodash/orderBy';
+import { Api_FinancialCode } from 'models/api/FinancialCode';
+import moment from 'moment';
+import React, { useCallback, useEffect, useMemo } from 'react';
+import { Col, Row } from 'react-bootstrap';
+import { FaPlus } from 'react-icons/fa';
+import { useHistory } from 'react-router';
+import styled from 'styled-components';
+
+import { useFinancialCodeRepository } from '../hooks/useFinancialCodeRepository';
+import FinancialCodeFilter, {
+  defaultFinancialCodeFilter,
+  IFinancialCodeFilter,
+} from './FinancialCodeFilter/FinancialCodeFilter';
+import { FinancialCodeResults } from './FinancialCodeResults/FinancialCodeResults';
+import * as Styled from './styles';
+
+/**
+ * Page that displays acquisition files information.
+ */
+export const FinancialCodeListView: React.FC = () => {
+  const history = useHistory();
+  const { hasRole } = useKeycloakWrapper();
+  const {
+    getFinancialCodes: {
+      execute: getFinancialCodes,
+      response: financialCodeResults,
+      loading: financialCodesLoading,
+    },
+  } = useFinancialCodeRepository();
+
+  const fetchData = useCallback(async () => {
+    await getFinancialCodes();
+  }, [getFinancialCodes]);
+
+  useEffect(() => {
+    if (financialCodeResults === undefined) {
+      fetchData();
+    }
+  }, [fetchData, financialCodeResults]);
```

Comment:
is this necessary? if the undefined check is removed this should only request once since fetchData isn't changing. Is there a time where we manually set financialCodeResults to undefined?

Reply:
Happy to change but was following pattern set in **ActivityList**

![image](https://user-images.githubusercontent.com/24322224/212771493-3a6beb83-43e9-47d5-9f0c-596ca7b45f19.png)